app/controllers/users_controller.rb

```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      redirect_to dashboard_path
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      render :new, obj: @user
+    end
+  end
+
+  private
+
+  def user_params
+    user_info = params.require(:user).permit(:name, :email, :password, :password_confirmation)
+    user_info[:email] = user_info[:email].downcase
```

Comment:
Looks great! Can we move `.downcase` into the model?

Reply:
yes!